src/d3a/models/config.py

```diff
@@ -86,6 +77,27 @@ def as_dict(self):
             if k in fields
         }
 
+    def event_update(self, cloud_coverage=None, pv_user_profile=None):
+        if cloud_coverage is not None:
+            self.read_cloud_coverage(cloud_coverage)
+        if pv_user_profile is not None:
+            self.read_pv_user_profile(pv_user_profile)
+
+    def read_cloud_coverage(self, cloud_coverage=0):
+        # TODO: Once the d3a uses a common API to the d3a-web, this should be removed
+        # since this limitation already exists on d3a-web
+        if 0 <= cloud_coverage <= 2:
+            self.cloud_coverage = cloud_coverage
+        else:
+            raise D3AException("Invalid cloud coverage value ({}).".format(cloud_coverage))
+
+    def read_pv_user_profile(self, pv_user_profile=None):
+        self.pv_user_profile = None \
+            if pv_user_profile is None \
```

Comment:
Why this check again? Isn't it performed in L84 already?

Reply:
`Yes, but this method is also called from line 49 (Config constructor)` Sorry, did not see that.
And yes, I agree: If we add more user profiles, we need to refactor.
Thanks!